File: cute-bedwars-database/src/main/java/io/github/gdrfgdrf/cutebedwars/database/base/IDatabase.java
package io.github.gdrfgdrf.cutebedwars.database.base;

/**
 * @author gdrfgdrf
 */
public interface IDatabase {
    String getDisplayName();
    void load() throws Exception;
    void close() throws Exception;

    <T> T getService(Class<? extends IService> serviceClass);
}

File: cute-bedwars-abstracts/src/main/java/io/github/gdrfgdrf/cutebedwars/abstracts/database/IDatabase.kt
package io.github.gdrfgdrf.cutebedwars.abstracts.database

import io.github.gdrfgdrf.multimodulemediator.Mediator
import io.github.gdrfgdrf.multimodulemediator.annotation.Service

@Service("database")
interface IDatabase {
    fun initialize()
    fun close()

    companion object {
        fun instance(): IDatabase = Mediator.get(IDatabase::class.java)!!
    }
}